book/javascript_modularity.rst

```diff
@@ -28,30 +54,72 @@ the following signature:
         definition function /*function for instantiating the module or object*/
     );
 
-.. note::
+In its simplest form, a RequireJS module is just an object defining pair-value pairs:
+
+.. code-block:: javascript
 
-    The ``module_id`` is typically equivalent to a folder path of simple packages,
-    so there is no need to define it.
+    // foo.js
 
-In the general case, the module definition looks as follows:
+    define({
+        foo: bar,
+        foobar: baz,
+        baz: function (param) {
+            // do something
 
-.. code-block:: javascript
+            return ...;
+        }
+    });
+
+    // or use a function if you need to do some initialization stuff:
+
+    define(function () {
```

Comment:
I would split this code-block on two separate. Because `define()` method (if it declares unnamed module) can be one per file-resource (if it is not build.js file, where all modules are named and loaded simultaneously). It might confuse of a reader. that `foo.js` file contains two `define()` calls with no `module_id` argument.